google/cloud/storage/lifecycle_rule.cc

```diff
@@ -141,11 +140,12 @@ void LifecycleRule::MergeConditions(LifecycleRuleCondition& result,
   }
   if (rhs.matches_storage_class.has_value()) {
     if (result.matches_storage_class.has_value()) {
-      std::vector<std::string> tmp;
-      std::vector<std::string> a = *std::move(result.matches_storage_class);
+      std::vector<std::string> a;
+      a.swap(*result.matches_storage_class);
       std::sort(a.begin(), a.end());
       std::vector<std::string> b = *rhs.matches_storage_class;
       std::sort(b.begin(), b.end());
+      std::vector<std::string> tmp;
       std::set_intersection(a.begin(), a.end(), b.begin(), b.end(),
                             std::back_inserter(tmp));
       result.matches_storage_class.emplace(std::move(tmp));
```

Comment:
The previous code w/ the move looked a bit more idiomatic, and likely how we'd write this if we wrote something similar again tomorrow. The only issue was re-using the moved-from `absl::optional` on line 151. I'd imagine another fix would be the previous code exactly as it was, but change line 151:

```diff
- result.matches_storage_class.emplace(std::move(tmp));
+ result.matches_storage_class = std::move(tmp);
```

Though any fix is fine w/ me, and I'll defer to your sensibilities.